res/controllers/Hercules-DJControl-Inpulse-500-scripts.js

```diff
@@ -0,0 +1,641 @@
+//Controller
+var INPULSE500 = {};
+
+//enables/disables vinyl mode
+var vinylEnabled = [
+    1,
+    1,
+];
+
+//This helps implement the beatmatch guide
+var leader = 0;
+
+//These store the timestamp when the song started playing
+var timestamps = [
+    Number.MAX_VALUE,
+    Number.MAX_VALUE,
+];
+
+//These store if the loop in and loop out markers are set
+var loopIn = 0;
+var loopOut = 0;
+
+//These store if the slicer mode is on
+var slicer = [
+    0,
+    0,
+];
+
+//These store the beat the song should be playing
+var slicerBeat = [
+    0,
+    0,
+];
+
+//These store the beat the song is playing
+var slicerBeatPlaying = [
+    0,
+    0,
+];
+
+//these store if the slicer needs to jump
+var slicerJump = [
+    false,
+    false,
+];
+
+//these store the position the slicer needs to jump to
+var slicerJumpPos = [
+    0,
+    0,
+];
+
+//these store if the slicer recently jumped
+var jumped = [
+    false,
+    false,
+];
+
+//rgb to colors index. these are the values sent by mixxx
+const rgb2index = {
+    12913160: 0,
+    3325508: 1,
+    4379892: 2,
+    16306688: 3,
+    17663: 4,
+    11469004: 5,
+    16557783: 6,
+    15921919: 7,
+    16744448: 8,
+};
+
+//the color conversion for pads is weird so using array to store values
+const colors = [
+    [64, 16, 18, 80, 2, 66, 78, 82, 76],
+    [96, 28, 31, 124, 3, 99, 123, 127, 116],
+];
```

Comment:
This will not work. The colors are arbitrary, as you can define any custom color scheme. Use the `ColorMapper` class instead, that will automatically map the RGB color sent by Mixxx to the closed predefined color available on the controller. 

See https://mixxx.org/news/2020-08-25-new-in-2-3-hotcue-colors/ for a high-level explanation. An example would look like this:

```js
let colors = new ColorMapper({
    0xFF0000: 1, // Red
    0x0000FF: 2, // Blue
    0xFFFF00: 3, // Yellow
    0x00FF00: 4, // Green
    // ...
});

// Get the color of hotcue 1. Lets assume that
// it's some shade of blue (similar but not equal
// to 0x0000FF.
let hotcueColor = engine.getValue("[Channel1]", "hotcue_1_color");
// Get the MIDI value to send to the controller
// that best matches the color.
let value = colors.getValueForNearestColor(hotcueColor);
// Variable `value` now equals 2
```

Reply:
actually this controller allows (almost) any rgb color to be displayed. Rewrote the logic a little bit to allow any color to be sent